### PR TITLE
mechanism let application skip apply the workload

### PR DESCRIPTION
**Pull request by wangyikewxgm**
**What this PR does / why we need it**:

add a traitDefintion parameter `manageWorkload`, to let application  skip apply the workload, this trait will take over it

related issue: #1605 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: apis/core.oam.dev/v1alpha2/core_types.go [hunk 1]
```diff
@@ -388,6 +388,10 @@ type ApplicationConfigurationComponent struct {
 	// Scopes in which the specified component should exist.
 	// +optional
 	Scopes []ComponentScope `json:"scopes,omitempty"`
+
+	// WorkloadManagedByTrait defines the workload managed by a trait
+	// +optional
+	WorkloadManagedByTrait bool `json:"manageByTrait,omitempty"`
```
Comment (wonderflow) [inline apis/core.oam.dev/v1alpha2/core_types.go:394]: why do we need to change spec on AppConfig?

Comment (wangyikewxgm) [reply apis/core.oam.dev/v1alpha2/core_types.go:394]: 因为我们每次assemble和dispatch其实是从appRev中拿到的数据，所以这个字段需要跟随ac被固化到apprev当中

Comment (wonderflow) [reply apis/core.oam.dev/v1alpha2/core_types.go:394]: 你现在改的是用户接口，不是实现。你只需要在实现上有这个字段，那就不能放到API上。

**Review by wonderflow — APPROVED** [review]
good job

**Pull request merged**